src/Sylius/Component/Product/spec/Model/ProductSpec.php

```diff
@@ -120,6 +122,66 @@ function it_refuses_to_remove_non_product_attribute(AttributeValueInterface $att
         $this->shouldThrow('\InvalidArgumentException')->duringRemoveAttribute($attribute);
     }
 
+    function it_returns_attributes_by_locale(
+        ProductAttributeInterface $attribute,
+        ProductAttributeValueInterface $attributeValueEN,
+        ProductAttributeValueInterface $attributeValuePL
+    ): void
+    {
+        $attribute->getCode()->willReturn('colour');
+
+        $attributeValueEN->setProduct($this)->shouldBeCalled();
+        $attributeValueEN->getLocaleCode()->willReturn('en_US');
+        $attributeValueEN->getAttribute()->willReturn($attribute);
+        $attributeValueEN->getCode()->willReturn('colour');
+        $attributeValueEN->getValue()->willReturn('Blue');
+
+        $attributeValuePL->setProduct($this)->shouldBeCalled();
+        $attributeValuePL->getLocaleCode()->willReturn('pl_PL');
+        $attributeValuePL->getAttribute()->willReturn($attribute);
+        $attributeValuePL->getCode()->willReturn('colour');
+        $attributeValuePL->getValue()->willReturn('Niebieski');
+
+        $this->addAttribute($attributeValueEN);
+        $this->addAttribute($attributeValuePL);
+
+        $this
+            ->getAttributesByLocale('pl_PL', 'en_US')
+            ->toArray()
+            ->shouldReturn([$attributeValuePL->getWrappedObject()])
```

Comment:
Btw. instead of `$this->getAttributesByLocale(...)->toArray()->shouldReturn([$mock->getWrappedObject()]);` you can use `$this->getAttributesByLocale(...)->shouldIterateAs([$mock]);`